distribution/lib/Standard/Base/0.0.0-dev/src/Data/List.enso

```diff
@@ -366,8 +366,9 @@ type List
         go list = case list of
             Nil -> Nothing
             Cons h t ->
-                f h
-                @Tail_Call go t
+                result = f h
+                if result.is_error then Error.throw result else
```

Comment:
```suggestion
                result.if_not_error <|
```

1. We should prefer the 'idiomatic' approach which is encapsulated by `if_not_error`
2. `Error.throw result` is actually wrong. It will likely work but more by accident - the thing is `result` is already a dataflow error, so calling `Error.throw already_error` just propagates the `already_error` without calling anything. It could have been `Function.identity already_error` and the behaviour would be the same 😅. If you want to use `Error.throw`, it'd probably have to be like `Error.throw result.catch` - first catch the payload and then re-throw it. But we don't want to do that as it would most likely reset the associated location/stack trace.

Reply:
Fixed